File: 2021/fsharp/Aoc2021.Tests/Day10.Tests.fs
module Day10.Tests

open Xunit

let input =
    [ "[({(<(())[]>[[{[]{<()<>>"
      "[(()[<>])]({[<{<<[]>>("
      "{([(<{}[<>[]}>{[]{[(<()>"
      "(((({<>}<{<{<>}{[]{[]{}"
      "[[<[([]))<([[{}[[()]]]"
      "[{[{({}]{}}([{[{{{}}([]"
      "{<[[]]>}<{[{[{[]{()[[[]"
      "[<(<(<(<{}))><([]([]()"
      "<{([([[(<>()){}]>(<<{{"
      "<{([{{}}[<[[[<>{}]]]>[]]" ]

[<Fact>]
let ``Day10 totalSyntaxErrorScore works`` () =
    Assert.Equal(26397, input |> totalSyntaxErrorScore)

[<Fact>]
let ``Day10 middleCompletion works`` () =
    Assert.Equal(288957L, input |> middleCompletion)

[<Fact>]
let ``Day10 Stars`` () =
    Assert.Equal("294195", job1 ())
    Assert.Equal("3490802734", job2 ())

File: 2021/csharp/Aoc2021.Tests/Day10Tests.cs
namespace Aoc2021.Tests;

public class Day10Tests
{
    private static readonly List<string> input = @"
        [({(<(())[]>[[{[]{<()<>>
        [(()[<>])]({[<{<<[]>>(
        {([(<{}[<>[]}>{[]{[(<()>
        (((({<>}<{<{<>}{[]{[]{}
        [[<[([]))<([[{}[[()]]]
        [{[{({}]{}}([{[{{{}}([]
        {<[[]]>}<{[{[{[]{()[[[]
        [<(<(<(<{}))><([]([]()
        <{([([[(<>()){}]>(<<{{
        <{([{{}}[<[[[<>{}]]]>[]]
    "
    .Trim()
    .Split('\n')
    .Select(line => line.Trim())
    .ToList();

    [Fact]
    public void TotalSyntaxErrorScoreWorks()
    {
        Assert.Equal(26397, Day10.TotalSyntaxErrorScore(input));
    }

    [Fact]
    public void MiddleCompletionScoreWorks()
    {
        Assert.Equal(288957, Day10.MiddleCompletionScore(input));
    }

    [Fact]
    public void Stars()
    {
        Day10 run = new();
        Assert.Equal("294195", run.Result1());
        Assert.Equal("3490802734", run.Result2());
    }
}
